### DISCUSSION TITLE
error ios code signing

### DISCUSSION THREAD

**Discussion by nicodemeMLG** [codemagic-ci-cd/codemagic-docs#2059] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [ ] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

build

### Expected results

i want to publish on appstore

### Actual results

No valid code signing certificates were found
You can connect to your Apple Developer account by signing in with your Apple ID
in Xcode and create an iOS Development Certificate as well as a Provisioning 
Profile for your project by:
  1- Open the Flutter project's Xcode target with
       open ios/Runner.xcworkspace
  2- Select the 'Runner' project in the navigator then the 'Runner' target
     in the project settings
  3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team. 
     You may need to:
         - Log in with your Apple ID in Xcode first
         - Ensure you have a valid unique Bundle ID
         - Register your device with your Apple Developer Account
         - Let Xcode automatically provision a profile for your app
  4- Build or run your project again
  5- Trust your newly created Development Certificate on your iOS device
     via Settings > General > Device Management > [your new certificate] > Trust

For more information, please visit:
  https://developer.apple.com/library/content/documentation/IDEs/Conceptual/
  AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html

Or run on an iOS simulator without code signing
════════════════════════════════════════════════════════════════════════════════
No development certificates available to code sign app for device deployment

Build failed :|
Failed to build for iOS

### Build id (optional)

65094ff5fc6e3e8df7a73e84

**Comment by dtrdic** [comment]
Hi @nicodemeMLG in order to publish your app to Appstore, you must first set up **iOS code signing** using a **distribution** code signing certificate, I suggest setting up automatic code signing as is the simplest approach, please follow [this guide](https://docs.codemagic.io/flutter-code-signing/ios-code-signing/#automatic-code-signing) in order to set it up.

After that just enable App Store Connect publishing using your App Store Connect API key
<img width="667" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/5939691b-c996-4a70-875c-a820a804d75b">

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by sandei-travolta** [reply]
done that but still experiencing the same issue
